java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -181,20 +183,66 @@ private FlowComputation(Set<SymbolicValue> symbolicValues, Predicate<Constraint>
     return flows;
   }
 
-  Stream<ExecutionPath> startPath(ExplodedGraph.Edge edge, PSet<Symbol> trackedSymbols) {
-    return new ExecutionPath(null, PCollections.emptySet(), trackedSymbols, ImmutableList.of(), false).addEdge(edge);
+  Stream<ExecutionPath> startPath(ExplodedGraph.Edge edge, PSet<Symbol> trackedSymbols, SameConstraints sameConstraints) {
+    return new ExecutionPath(null, PCollections.emptySet(), trackedSymbols, sameConstraints, ImmutableList.of(), false).addEdge(edge);
+  }
+
+  private static class SameConstraints {
+    private final Map<Symbol, Boolean> sameConstraintsBySymbol;
```

Comment:
Instead of `Map<Symbol, Boolean>` you can use `PSet<Symbol>`  where only symbols with same constraints are kept. Then you can test with `contains`. Also PSet is persistent, so you can share it without copying.